applications/cryptex.md

```diff
@@ -0,0 +1,295 @@
+
+### Milestone 2 — Encryption to Future Slots
+
+- **Estimated Duration:** 1 month
+- **FTE:**  1.5
+- **Costs:** 10,000 USD
+
+Goal: We want to enable encryption to future slots, including slot scheduling, encryption, and decryption. The following items also necesitates the development of a basic user interface, which uses the light client and SDK which we develop.
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 1. | Light Client | We implement a light client (based on smoldot) with the added functionality that it: a. can open connections to specific nodes b. ensure clocks are properly set, otherwise return an error. This is to ensure proper synchronization, so that slot scheduling can be reliable/accurate. |
+| 2 | User Interface: setup | We introduce a user interface which will act as a testbed for integrations between the light client and the SDK. The user interface will be a React project, will connect to the network via the light client, and will interface with IPFS (for storage and retrieval of ciphertexts). This intention is to integrated both light client and SDK and also to ensure that interactions with the chain function as intended. |
+| 3. | SDK: Slot Scheduling | We implement slot scheduling logic to identify a future slot based on some future 'time' and derive its inputs. |
+| 4. | SDK: Encryption | Using the output of the slot scheduler, the user agent will be able to encrypt to a future slot or epoch. Ciphertexts will be stored offchain in IPFS, and we will refer to stored ciphertexts by their CID. |
+| 5. | SDK: Decryption | After a block is authored for the specified future slot, we can decrypt the secret by fetching the secret published with the block (if encrypted to a slot)  or a threshold of published secrets (encrypted to epoch) and using it to decrypt the ciphertext created previously. |
+ 
+### Milestone 3: Putting it all together - Sealed-Bid NFT Auction PoC
+
+- **Estimated Duration:** 1 month
+- **FTE:**  1.5
+- **Costs:** 10,000 USD
+
+Goal: We want to put everything developed thus far into action by developing an application that runs on the EtF network. A simple use case is to create a sealed-bid auction platform, where bids are sealed until a certain slot in the future. This work will include the development of an auction contract along with a user interface. At the end of this milestone, we will have a small testnet hosted on google cloud as well as a dapp for sealed-bid NFT auctions.
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 1. | Smart Contract: Auction Contract | We develop a smart contract (and host it on our blockchain) that enables an on-chain auction where participants can issue a sealed-bid which will unseal at some specific future slot. The sealed bid is stored in the contract. |
```

Comment:
This is written in ink!, I assume? That means your chain supports the `contracts` pallet?

Reply:
Yes, written in ink! and using the contracts pallet. I didn't include it explicitly, but as a basis we'll use Grandpa, our modified Aura, Contracts, Balances, Timestamp, insecure-randomness-collective-flip, and possibly the Assets pallet if we see a need (for the final milestone).